include/igl/readPLY.cpp

```diff
@@ -38,140 +39,313 @@ template <
 IGL_INLINE bool igl::readPLY(
-  FILE * ply_file,
+  std::istream & ply_stream,
   std::vector<std::vector<Vtype> > & V,
   std::vector<std::vector<Ftype> > & F,
   std::vector<std::vector<Ntype> > & N,
-  std::vector<std::vector<UVtype> >  & UV)
-{
-  using namespace std;
-   typedef struct Vertex {
-     double x,y,z;          /* position */
-     double nx,ny,nz;         /* surface normal */
-     double s,t;              /* texture coordinates */
-     void *other_props;       /* other properties */
-   } Vertex;
-
-   typedef struct Face {
-     unsigned char nverts;    /* number of vertex indices in list */
-     int *verts;              /* vertex index list */
-     void *other_props;       /* other properties */
-   } Face;
-
-  igl::ply::PlyProperty vert_props[] = { /* list of property information for a vertex */
-    {"x", PLY_DOUBLE, PLY_DOUBLE, offsetof(Vertex,x), 0, 0, 0, 0},
-    {"y", PLY_DOUBLE, PLY_DOUBLE, offsetof(Vertex,y), 0, 0, 0, 0},
-    {"z", PLY_DOUBLE, PLY_DOUBLE, offsetof(Vertex,z), 0, 0, 0, 0},
-    {"nx", PLY_DOUBLE, PLY_DOUBLE, offsetof(Vertex,nx), 0, 0, 0, 0},
-    {"ny", PLY_DOUBLE, PLY_DOUBLE, offsetof(Vertex,ny), 0, 0, 0, 0},
-    {"nz", PLY_DOUBLE, PLY_DOUBLE, offsetof(Vertex,nz), 0, 0, 0, 0},
-    {"s", PLY_DOUBLE, PLY_DOUBLE, offsetof(Vertex,s), 0, 0, 0, 0},
-    {"t", PLY_DOUBLE, PLY_DOUBLE, offsetof(Vertex,t), 0, 0, 0, 0},
-  };
-
-  igl::ply::PlyProperty face_props[] = { /* list of property information for a face */
-    {"vertex_indices", PLY_INT, PLY_INT, offsetof(Face,verts),
-      1, PLY_UCHAR, PLY_UCHAR, offsetof(Face,nverts)},
-  };
-
-  int nelems;
-  char ** elem_names;
-  igl::ply::PlyFile * in_ply = igl::ply::ply_read(ply_file,&nelems,&elem_names);
-  if(in_ply==NULL)
+  std::vector<std::vector<UVtype> > & UV)
+{   
+  igl::tinyply::PlyFile file;
+  file.parse_header(ply_stream);
+
+  /*
+  std::cout << "igl::readPLY Header..........................................................\n";
+  for (auto c : file.get_comments())
+      std::cout << "Comment: " << c << std::endl;
+  for (auto e : file.get_elements())
   {
-    return false;
+      std::cout << "element - " << e.name << " (" << e.size << ")" << std::endl;
+      for (auto p : e.properties)
+          std::cout << "\tproperty - " << p.name << " (" << igl::tinyply::PropertyTable[p.propertyType].str << ")" << std::endl;
+  }
+  std::cout << "igl::readPLY Data..........................................................\n";
+  //*/
+
+  // Tinyply treats parsed data as untyped byte buffers.
+  std::shared_ptr<igl::tinyply::PlyData> vertices, normals, faces, texcoords;
+
+  // The header information can be used to programmatically extract properties on elements
+  // known to exist in the header prior to reading the data. For brevity of this sample, properties
+  // like vertex position are hard-coded:  
+  try {
+    vertices = file.request_properties_from_element("vertex", { "x", "y", "z" });
   }
+  catch (const std::exception & e) { }
```

Comment:
Shouldn't we return `false` if an exception occurs?

Reply:
This is the suggested way to ask if a .ply file contains (or not) a property associate to vertex. 
In this case, we could argument that if a file does not contain (x,y,z) properties in vertex, then you can not really store a mesh. Also, somebody could argument that "an empty mesh is still a mesh", so I would let that code as is.

Reply:
Ah, I see. I don't really like that it uses exceptions to signal the absence of optional attributes, but I guess that's how it is.